core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -347,15 +353,16 @@ public void onLeadershipChange(Set<Partition> partitionsBecomeLeader,
     public void stopPartitions(TopicPartition topicPartition, boolean delete) {
         if (delete) {
             // Delete from internal datastructures only if it is to be deleted.
-            Uuid topicIdPartition = topicPartitionIds.remove(topicPartition);
-            LOGGER.debug("Removed partition: {} from topicPartitionIds", topicIdPartition);
+            Map<String, Uuid> mapping = metaDataCache.topicNamesToIds();
+            Uuid topicIdPartition = mapping.remove(topicPartition.topic());
```

Comment:
The idea of replacing this cache with Metadata cache is that the existing metadata cache will serve as the source of truth. It should already be updated to contains the correct topics Id <-> topicPartition mapping irrespective of whether tiered storage is turned on or not. TS code should only access this cache and not modify it in any way. The modification (adding/removal) should already exist is rest of the normal path Kafka code. (If it doesn't then we need to fix it).

Reply:
Got it. So can I say for `stopPartition` and `onLeadershipChange`, there's no need to update the metadatacache at all ? Is there any extra logic that should be addded into these 2 APIs ?

Reply:
I don't think there is anything to be done here. The topic partitions should disappear once the controller lets us know they should.

Reply:
@clolov Thanks a lot for your comment. I've incorporated your comments